src/evaluate/evaluator.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2022 The HuggingFace Datasets Authors and the TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from numbers import Number
+from typing import Dict, Optional, Union
+
+# Lint as: python3
+import numpy as np
+from datasets import Dataset, DatasetDict, IterableDataset, IterableDatasetDict, load_dataset, load_metric
+from datasets.metric import Metric
+from scipy.stats import bootstrap
+from transformers import Pipeline, pipeline
+
+
+class Evaluator:
+    def __init__(
+        self,
+        pipe: Union[str, Pipeline],
```

Comment:
This should also include a `Callable` type for users that setup their own pipeline (e.g. for a `sklearn` model).